### ISSUE TITLE
[MNG-7248] Fix warnings issued by dependency:analyze

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8195]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** opened **[MNG-7248](https://issues.apache.org/jira/browse/MNG-7248?redirect=false)** and commented

At least the following needs to be addressed:

```
 501 [INFO] --- maven-dependency-plugin:3.1.1:analyze (default-cli) @ maven-model-builder ---
 502 [WARNING] Used undeclared dependencies found:
 504 [WARNING]    org.apache.commons:commons-lang3:jar:3.12.0:compile

 776 [INFO] --- maven-dependency-plugin:3.1.1:analyze (default-cli) @ maven-resolver-provider ---
 777 [WARNING] Used undeclared dependencies found:
 779 [WARNING]    org.apache.maven:maven-builder-support:jar:4.0.0-alpha-1-SNAPSHOT:compile
 780 [WARNING]    org.apache.maven:maven-artifact:jar:4.0.0-alpha-1-SNAPSHOT:compile
 781 [WARNING] Unused declared dependencies found:
 782 [WARNING]    com.google.guava:guava:jar:30.1-jre:compile
 783 [WARNING]    com.google.guava:failureaccess:jar:1.0.1:compile
 787 [WARNING]    org.slf4j:slf4j-api:jar:1.7.32:compile

 875 [INFO] --- maven-dependency-plugin:3.1.1:analyze (default-cli) @ maven-core ---
 876 [WARNING] Used undeclared dependencies found:
 877 [WARNING]    org.slf4j:slf4j-api:jar:1.7.32:compile
 878 [WARNING]    org.codehaus.plexus:plexus-interpolation:jar:1.26:compile
 882 [WARNING]    org.codehaus.plexus:plexus-component-annotations:jar:2.1.0:compile
 883 [WARNING] Unused declared dependencies found:
 884 [WARNING]    com.google.guava:guava:jar:30.1-jre:compile
 885 [WARNING]    com.google.guava:failureaccess:jar:1.0.1:compile

1089 [INFO] --- maven-dependency-plugin:3.1.1:analyze (default-cli) @ maven-embedder ---
1093 [WARNING] Unused declared dependencies found:
1094 [WARNING]    com.google.guava:guava:jar:30.1-jre:compile
1095 [WARNING]    com.google.guava:failureaccess:jar:1.0.1:compile
1096 [WARNING]    javax.annotation:jsr250-api:jar:1.0:compile

1181 [INFO] --- maven-dependency-plugin:3.1.1:analyze (default-cli) @ maven-compat ---
1182 [WARNING] Used undeclared dependencies found:
1183 [WARNING]    javax.inject:javax.inject:jar:1:compile
```

---

**Affects:** 3.0-alpha-1

**Remote Links:**
- [GitHub Pull Request #546
](https://github.com/apache/maven/pull/546)

**Backported to:** [4.0.0-alpha-2](https://github.com/apache/maven/milestone/113?closed=1)

**Comment by jira-importer** [comment]
**[Guoxiong Li](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=liguoxiong)** commented

Hi Michael Osipov,

 

I am new to the maven project and want to solve this issue as a start.

Did you begin the work locally? If not, I would like to take it over and try.

 

Best Regards,

- Guoxiong

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

No, I have addressed in 3.8.x only. If you want to work on it -- great. Try to understand the issue, provide a PR and I will review it.

**Comment by jira-importer** [comment]
**[Guoxiong Li](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=liguoxiong)** commented

I submitted the PR just now. Thanks for taking the time to review.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [5fd536853fc945c63b267e2a92703d9b7bc8888b](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=5fd536853fc945c63b267e2a92703d9b7bc8888b).
